src/main/java/com/code_intelligence/jazzer/api/BugDetectors.java

```diff
@@ -0,0 +1,111 @@
+   *
+   * <p>By wrapping the call into a try-with-resources statement, network connection permissions
+   * can be configured to apply to individual parts of the fuzz test only:
+   *
+   * <pre>{@code
+   *   Image image = parseImage(bytes);
+   *   Response response;
+   *   try (SilentCloseable unused = BugDetectors.allowNetworkConnections()) {
+   *     response = uploadImage(image);
+   *   }
+   *   handleResponse(response);
+   * }</pre>
+   *
+   * @return a {@link SilentCloseable} that restores the previously set permissions when closed
+   */
+  public static SilentCloseable allowNetworkConnections() {
+    return allowNetworkConnections((host, port) -> true);
+  }
+
+  /**
+   * Allows all network connections for which the provided predicate returns {@code true}.
+   *
+   * <p>By default, all attempted network connections are considered unexpected and result in a
+   * finding being reported.
+   *
+   * <p>By wrapping the call into a try-with-resources statement, network connection permissions
+   * can be configured to apply to individual parts of the fuzz test only:
+   *
+   * <pre>{@code
+   *   Image image = parseImage(bytes);
+   *   Response response;
+   *   try (SilentCloseable unused = BugDetectors.allowNetworkConnections(
+   *       (host, port) -> host.equals("example.org"))) {
+   *     response = uploadImage(image, "example.org");
+   *   }
+   *   handleResponse(response);
+   * }</pre>
+   *
+   * @param connectionPermitted a predicate that evaluate to {@code true} if network connections to
+   *                            the provided combination of host and port are permitted
+   * @return a {@link SilentCloseable} that restores the previously set predicate when closed
+   */
+  public static SilentCloseable allowNetworkConnections(
+      BiPredicate<String, Integer> connectionPermitted) {
+    if (connectionPermitted == null) {
+      throw new IllegalArgumentException("connectionPermitted must not be null");
+    }
+    if (currentPolicy == null) {
+      throw new IllegalStateException("Failed to set network connection policy");
+    }
+    BiPredicate<String, Integer> previousPolicy = currentPolicy.getAndSet(connectionPermitted);
+    return () -> {
+      if (!currentPolicy.compareAndSet(connectionPermitted, previousPolicy)) {
+        throw new IllegalStateException(
+            "Failed to reset network connection policy - using try-with-resources is highly recommended");
+      }
+    };
+  }
+
+  private static AtomicReference<BiPredicate<String, Integer>> getConnectionPermittedReference() {
+    try {
+      Class<?> ssrfSanitizer =
+          Class.forName("com.code_intelligence.jazzer.sanitizers.ServerSideRequestForgery");
+      return (AtomicReference<BiPredicate<String, Integer>>) ssrfSanitizer
+          .getField("connectionPermitted")
+          .get(null);
+    } catch (ClassNotFoundException | NoSuchFieldException | IllegalAccessException e) {
+      System.err.println("WARNING: ");
```

Comment:
Use `Log.warn`?

Reply:
We can't in sanitizers as `Log` is not (and probably shouldn't be) in the API package.

Reply:
Ah, yes, right!